tests/InlineTextTest.php

```diff
@@ -27,29 +24,53 @@ public static function expectedRenders()
                 new InlineText("  Hello world!  ", "¯\_(ツ)_/¯"),
                 "  ¯\_(ツ)_/¯Hello world!¯\_(ツ)_/¯  "
             ),
-            "expect only whitespace and subtype" => array(
+            "expect text with internal newlines" => array(
+                new InlineText("First\nSecond\n\n\nThird", ":"),
+                ":First:\n:Second:\n\n:Third:"
+            ),
+            "expect only whitespace with subtype" => array(
                 new InlineText("   ", "¯\_(ツ)_/¯"),
                 "   "
             ),
-            "expect text with internal newline and trailing whitespace" => array(
-                new InlineText("First\nSecond ", ":"),
-                ":First:\n:Second: "
+            "expect empty string" => array(
+                new InlineText(""),
+                ""
+            ),
+            "expect empty string with subtype" => array(
+                new InlineText("", "~"),
+                ""
+            ),
+            "expect multiple lines of whitespace only to be preserved with spaces on the first line" => array(
+                new InlineText("  \n"),
+                "\n"
+            ),
+            "expect multiple lines of whitespace only to be preserved with spaces on the last line" => array(
```

Comment:
I don't fully understand this test. Could it be named another way?

Reply:
@graycodes Yeah, this is poorly named, taking a look now. (I named hundreds of tests in a short time period so I didn't really think them through!)